src/PValueBasedFeatureSelector.jl

```diff
@@ -0,0 +1,110 @@
+"""
+`PValueBasedFeatureSelector` has the following fields:
+
+* `k::Int64` - Select top `k` features with the highest correlation to target
+  variable. You could ignore this by specifying k <= 0. This defaults to 0.
+* `threshold::Float64` - Select features with correlation less than or equal to
+  threshold. Note that, in P-value, a feature is considered important when the value
+  is close to 0. To ignore, simply set threshold to 0 (default behavior).
```

Comment:
For fields which are optional the common convention would be to use `nothing` rather than having a special sentinel value which you need to carefully document.

It should be quite ok performance-wise to use `Union{T,Nothing}` as the field type, (`k::Union{Int64,Nothing}` etc).